project.yaml

```diff
@@ -907,14 +907,43 @@ Layer:
               ('railway_' || (CASE WHEN railway IN ('platform') THEN railway ELSE NULL END)),
               (('aeroway_' || CASE WHEN aeroway IN ('runway', 'taxiway', 'helipad') THEN aeroway ELSE NULL END))
             ) AS feature
-          FROM planet_osm_polygon
-          WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
-            OR railway IN ('platform')
-            OR aeroway IN ('runway', 'taxiway', 'helipad')
+          FROM (
+          SELECT 
+              way, highway, railway, aeroway, z_order, way_area FROM planet_osm_polygon
+            WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
+              OR railway IN ('platform')
+              OR aeroway IN ('taxiway', 'helipad')
+          UNION ALL
+          SELECT -- runways with width either tagged or estimated from runway length
+              ST_Buffer(
+                way,
+                0.5*(St_Distance(ST_StartPoint(way), ST_EndPoint(way))/St_Distance_Sphere(ST_Transform(ST_StartPoint(way), 4326), ST_Transform(ST_EndPoint(way), 4326)))*
+                  CASE
+                    WHEN width ~ '^-?\d{1,4}(\.\d+)?$' THEN width::NUMERIC
```

Comment:
I considered that but i don't see the gain (except for slightly shortening the code) and it would only work on web mercator.  I suppose computationally ST_Length(way::geography) is identical to ST_Length_Spheroid().

Ideally mapnik should make available the tile center scale factor in a similar way as !pixel_width!.